src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1316,4 +1403,85 @@ private void setField(CpsFlowExecution result, String fieldName, Object value) {
      * this field is set to {@link CpsFlowExecution} that will own it.
      */
     static final ThreadLocal<CpsFlowExecution> PROGRAM_STATE_SERIALIZATION = new ThreadLocal<CpsFlowExecution>();
+
+    class TimingFlowNodeStorage extends FlowNodeStorage {
+        private final FlowNodeStorage delegate;
```

Comment:
🎆  so glad this is a raw FlowNodeStorage -- this means optimized FlowNodeStorage can use this bad boy.